doc/classes/StringName.xml

```diff
@@ -5,7 +5,7 @@
 	</brief_description>
 	<description>
 		[StringName]s are immutable strings designed for general-purpose representation of unique names (also called "string interning"). Two [StringName]s with the same value are the same object. Comparing them is extremely fast compared to regular [String]s.
-		You will usually just pass a [String] to methods expecting a [StringName] and it will be automatically converted, but you may occasionally want to construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code].
+		You will usually pass a [String] to methods expecting a [StringName] and it will be automatically converted (often at compile time), but in rare cases you can construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code]. Manually constructing a [StringName] allows you to control when conversions occur (to move them out of loops) or eliminate them (with the literal).
```

Comment:
My previous suggestion was a way to keep what you were trying to convey in a way that flows more nicely. I understand your point but I think is worded a bit awkwardly at the moment.

```suggestion
You will usually pass a [String] to methods expecting a [StringName] and it will be automatically converted (often at compile time), but in rare cases you can construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code]. Manually constructing a [StringName] allows you to control when the conversion from [String] occurs or prevent it entirely (with the literal), which can be useful for loops or other repetitive operations.
```

Reply:
In that case, pull back even more on making it sound useful and simplify further:
>Manually constructing a [StringName] allows you to control when the conversion from [String] occurs or to use the literal and prevent conversions entirely.

Reply:
Sounds fair enough to me

Reply:
The last proposal sounds good to me, could you amend the commit with that?

Reply:
Did so myself in the end.